src/main/java/org/opentripplanner/raptor/rangeraptor/standard/stoparrivals/view/StopsCursor.java

```diff
@@ -185,6 +186,11 @@ private ArrivalView<T> newAccessView(
   ) {
     // Get the real 'departureTime' honoring the time-shift restriction in the access
     int departureTime = transitCalculator.departureTime(accessPath, preferredDepartureTime);
+
+    if (departureTime == SearchParams.TIME_NOT_SET) {
+      throw new IllegalStateException("The departureTime is not found");
+    }
```

Comment:
I did test the performance on this, it should not have an effect and the test also indicate that. But the commit as a hole has slightly worse performance than the previous commit. But it is hard to measure, and any reliable times would need a lot more samples than I run.

Reply:
Looks ok to me